singer/src/main/java/com/pinterest/singer/writer/KafkaWritingTask.java

```diff
@@ -51,18 +53,25 @@ public KafkaWritingTask(KafkaProducer<byte[], byte[]> producer,
     this.taskCreationTimeInMillis = System.currentTimeMillis();
     try {
       leaderNode = sortedPartitions.get(msgs.get(0).partition()).leader().host();
+      partition = sortedPartitions.get(msgs.get(0).partition()).partition();
```

Comment:
update the two lines above to reduce redundant function calls?

        PartitionInfo firstPartition = sortedPartitions.get(msgs.get(0).partition());
        leaderNode = firstPartition.leader().host();
        partition = firstPartition.partition();

Reply:
made the change in the update of the PR.